ansible/deploy_riakcs_new.yml

```diff
@@ -0,0 +1,197 @@
+# This is a temporary deployment script for managing hosts in [riakcs_new],
+# which is a temporary group used while migrating to a new riak cluster.
+#
+# This is a copy of deploy_riakcs.yml with replacements:
+# - hosts:
+#   riakcs -> riakcs_new
+#   stanchion -> stanchion_new
+#   riakcs_control_host -> riakcs_new_control_host
+#
+# Also added one task to set riakcs_control and updated variable assignments.
+#
+# run with --tags=mount-ecryptfs to mount encrypted volume after reboot
+
+- name: Set variables
+  hosts:
+    - riakcs_new
+    - stanchion_new
+  tasks:
+    - set_fact: _riak_backend="{{ riak_new_backend }}"
+    - set_fact: _riak_ring_size="{{ riak_new_ring_size }}"
+    - set_fact: riakcs_control="{{ groups.riakcs_new[0] }}"
+    - set_fact: stanchion_host="{{ groups.stanchion_new[0] }}"
+
+- name: Common Setup
+  hosts:
+    - riakcs_new
+    - stanchion_new
+  become: true
+  roles:
+    - {role: ecryptfs}
+
+- name: First host in riakcs_new group -> riakcs_new_control_host
+  # riakcs_new_control_host group must contain at most one host
+  hosts:
+    - riakcs_new
+  tasks:
+    - add_host: name="{{ riakcs_control }}" groups=riakcs_new_control_host
+      when: riakcs_new_control_host is undefined
+      check_mode: no
+      changed_when: false
+      tags: localsettings
+
+- name: Register riak conf override fact
+  # NOTE riak config files are only created/updated if they do not already exist
+  # or if '-e force_riak_config=host1,host2,...' is specified on the command
+  # line listing each node to force configure (use inventory_hostname as host
+  # name). This is because it can be dangerous to change the riak.conf of a
+  # running node. The Riak service may refuse to start or ignore some data if
+  # the configuration is changed.
+  hosts:
+    - riakcs_new
+  vars_files:
+    - roles/riakcs/install/defaults/main.yml
+    - roles/riakcs/configure/vars/main.yml
+  tasks:
+    - name: "Stat {{ riak_conf }}"
+      stat: path="{{ riak_conf }}"
+      register: riak_conf_file
+      changed_when: false
+      failed_when: false
+      check_mode: no
+    - set_fact: _force_riak_config="{{ 'yes'
+        if inventory_hostname in (force_riak_config | default('')).split(',')
+            or force_riak_config | default('') == 'all'
+            or not (riak_conf_file.stat.exists | default(True))
+        else '' }}"
+    # HACK set fact because defaults cannot be loaded from other hosts
+    - set_fact: riak_nodename="{{ backend_vars.riak_nodename }}"
+
+- name: Configure swap for riak hosts
+  hosts: riakcs_new
+  roles:
+    - {role: swap}
+
+- name: Load admin keys from Riak CS conf
+  become: true
+  hosts: riakcs_new_control_host
+  vars_files:
+    - roles/riakcs/install/defaults/main.yml
+  tasks:
+    - include: roles/riakcs/get_admin_keys.yml tags=localsettings
```

Comment:
Not neccessary now but could make a custom fact file to get these: http://serverascode.com/2015/01/27/ansible-custom-facts.html